### ISSUE TITLE
feature request: optional list options

### ISSUE THREAD

**Issue by spalberg** [pforret/bashew#2]
Hi @pforret,
bashew is a really great template and it saves me a lot of time!

I was wondering if you could add support for optional list options.
As an example I can present the `-f` option of the `docker-compose` CLI command.
It is optional but you can pass it multiple times to source different `docker-compose`-files:
```
docker-compose [-f <arg>...] [options] [COMMAND] [ARGS...] 
docker-compose up # works!
docker-compose -f docker-compose.yml -f docker-compose.admin.yml up # works too!
```

I can achieve something similar using bashew when I pass the options as a single string:
```
bashew-compose -f "docker-compose.yml docker-compose.admin.yml" up
```
However, doing it this way I lose the benefit of filename completion by my shell.

Do you think this would be a useful feature?
Is there an easy way to add this to bashew?

**Comment by pforret** [comment]
Hi Sven,
what you want, is options that can be called > 1. This is not easy because of the way the options are parsed in bashew: 
`--config [value]` is parsed into a $config variable, not an array.
A possible solution would be to allow `--config:1 [value] --config:2 [value]` syntax which would then be parsed into an array.

Me, I currently use comma-separated values e.g. `--effect dark,blur,grain` in [pforret/splashmark](https://github.com/pforret/splashmark), but you're right that it doesn't work well for file names.

It's a good suggestion, let me think on that one!

**Comment by usrme** [comment]
First off, kudos to you, @pforret, for creating such a wonderful project! The amount of effort you've put in really shines through. 

Secondly, I'm quite possibly showing my own ignorance here as I don't know the intricacies of how Bashew does parameter parsing, but why, in this case, it isn't possible to simply define an array variable for a given parameter and append to it while going over the parameters?

Here's a simple working example outside of Bashew that on the face of it seems to work:

```shell
#!/usr/bin/env bash

script_name=$(basename "${BASH_SOURCE[0]}")
script_ver='0.0.1'

usage() {
  cat << EOF
Usage: ${script_name} [-h | --help] [-v | --verbose] [-V | --version]

Example script to demonstrate for optional list options.

Available options:

-h, --help           Print this help and exit
-v, --verbose        Print script debug info
-V, --version        Print version number and exit
-c, --config-file    Configuration file(s) to be consumed

Example: ${script_name} -c file1.yml -c file2.yml
Example: ${script_name} --config-file file1.yml --config-file file2.yml

EOF
  exit
}

config_files=()

while :; do
  case "$1" in
  -h | --help) usage ;;
  -v | --verbose)
    set -x
    export SHELLOPTS
    ;;
  -V | --version) echo "$script_ver" && exit ;;
  -c | --config-file)
    config_files+=("$2")
    shift
    ;;
  ?*)
    echo "Unexpected parameter: '${1}'"
    usage
    ;;
  *) break ;;
  esac
  shift
done

for file in "${config_files[@]}"; do
  echo "Working with file: ${file}"
done
```

**Comment by pforret** [comment]
Hi @usrme , 

you are exactly right that the implementation should be done with bash arrays - `config_files+=("$2")`

It's a bit more involved in **bashew** because I use just 1 function `list_options()` to define all flags/options/parameters and all usage info/variable initialisation/parsing is derived from there. I just implemented this in **bashew** version `1.13.0` (the 'normal' template)

Hi @spalberg ,

I did it the following way: there is now an option type 'list' that is by default an empty array and that you can call multiple times: `--file /path/file1 --file /path/file2` and that you can use after automatic parsing as an bash array `${file[@]}`

Thanks for the suggestion and I hope this makes **bashew** even better for your usage!

Peter
